build/uploadperftobenchview.sh

```diff
@@ -0,0 +1,115 @@
+            exit 1
+        fi
+    else
+            echo EnvVar runType should be set; exiting...
+            exit 1
+    fi
+fi
+if ! env | grep -q ^GIT_BRANCH=
+then
+    echo EnvVar GIT_BRANCH should be set; exiting...
+    exit 1
+fi
+
+rm -r -f $perfWorkingDirectory/Microsoft.BenchView.JSONFormat > /dev/null 2>&1
+
+if [ ! -d $perfWorkingDirectory/Microsoft.BenchView.JSONFormat ]; then
+    # curl "http://benchviewtestfeed.azurewebsites.net/nuget/FindPackagesById()?id='Microsoft.BenchView.JSONFormat'" | grep "content type" | sed "$ s/.*src=\"\([^\"]*\)\".*/\1/;tx;d;:x" | xargs curl -o $perfWorkingDirectory/benchview.zip http://benchviewtestfeed.azurewebsites.net/api/v2/package/microsoft.benchview.jsonformat/0.1.0-pre001
+
+    curl -o $perfWorkingDirectory/benchview.zip http://benchviewtestfeed.azurewebsites.net/api/v2/package/microsoft.benchview.jsonformat/0.1.0-pre024
+    unzip -q -o $perfWorkingDirectory/benchview.zip -d $perfWorkingDirectory/Microsoft.BenchView.JSONFormat
+fi
+# nuget install Microsoft.BenchView.JSONFormat -Source http://benchviewtestfeed.azurewebsites.net/nuget -OutputDirectory $perfWorkingDirectory -Prerelease -ExcludeVersion || { echo Failed to install Microsoft.BenchView.JSONFormat NuPkg && exit 1 ; }
+
+# Do this here to remove the origin but at the front of the branch name as this is a problem for BenchView
+if [[ "$GIT_BRANCH" == "origin/"* ]]
+then
+    GIT_BRANCH_WITHOUT_ORIGIN=${GIT_BRANCH:7}
+else
+    GIT_BRANCH_WITHOUT_ORIGIN=$GIT_BRANCH
+fi
+
+benchViewName="SDK perf $OS $architecture $configuration $runType $GIT_BRANCH_WITHOUT_ORIGIN"
+if [[ "$runType" == "private" ]]
+then
+    benchViewName="$benchViewName $BenchviewCommitName"
+fi
+if [[ "$runType" == "rolling" ]]
+then
+    benchViewName="$benchViewName $GIT_COMMIT"
+fi
+echo BenchViewName: $benchViewName
+
+echo Creating: $perfWorkingDirectory/submission-metadata.json
+python3.5 "$perfWorkingDirectory/Microsoft.BenchView.JSONFormat/tools/submission-metadata.py" --name "$benchViewName" --user-email "dotnet-bot@microsoft.com" \
+                    -o "$perfWorkingDirectory/submission-metadata.json" || { echo Failed to create: $perfWorkingDirectory/submission-metadata.json && exit 1 ; }
+
+echo Creating: $perfWorkingDirectory/build.json
+python3.5 "$perfWorkingDirectory/Microsoft.BenchView.JSONFormat/tools/build.py" git --branch $GIT_BRANCH_WITHOUT_ORIGIN --type "$runType" \
+                   -o "$perfWorkingDirectory/build.json" || { echo Failed to create: $perfWorkingDirectory/build.json && exit 1 ; }
+
+echo Creating: $perfWorkingDirectory/machinedata.json
+python3.5 "$perfWorkingDirectory/Microsoft.BenchView.JSONFormat/tools/machinedata.py" \
+                   -o "$perfWorkingDirectory/machinedata.json" || { echo Failed to create: $perfWorkingDirectory/machinedata.json && exit 1 ; }
+
+echo Creating: $perfWorkingDirectory/measurement.json
+find $perfWorkingDirectory -maxdepth 1 -type f -name "*.xml" -exec echo Processing: "{}" \;
```

Comment:
Where are the result XML files being dropped and where are we running this command from? I assume the command is executed from the workspace root, but want to make sure. I think it might be good here to do some kind of error checking that we are still finding the results and making sure that we are generating the measurement.json with all of the data.

Reply:
"making sure that we are generating the measurement.json" - we do a check at each call to "measurement.py" Is that not sufficient?

Reply:
It's more I am worried that we could change the way that the results are produced and now the find command does not return all of the results to be processed. Given this is likely a small likelihood, but just something that I noticed.